### ISSUE TITLE
[ENH] `make_reduction` reducers should allow parsing the values as pandas dataframe to its estimator.

### ISSUE THREAD

**Issue by kdekker-kdr4** [alan-turing-institute/sktime#8755]
**Is your feature request related to a problem? Please describe.**
Allow to parse the values to the estimator as pandas dataframe.
Some attributes on sklearn estimators are only set if the input is pandas dataframe and are not set during fit with the current approach of using numpy array.

For example the attribute "feature_names_in_". E.g. useful to link features to potential feature importance generated during fit as in the example below.

```python
from sktime.forecasting.compose import make_reduction
from sktime.datasets import load_airline
from sklearn.ensemble import RandomForestRegressor
y = load_airline()
regressor = RandomForestRegressor()
forecaster = make_reduction(regressor, window_length=15, strategy="recursive")
forecaster.fit(y)
forecaster.estimator_.feature_names_in_ #Not working now.
```

**Describe the solution you'd like**
Allow to set input type similar to TabularToSeriesAdaptor.

**Additional context**
I imagine this could be of value for the other estimator classes that can be created via the `make_reduction` function.

**Comment by fkiraly** [comment]
`RecursiveRegressionForecaster` does allow this is directly called! It is the "2nd generation", but `make_reduction` still points to the first iteration which converts to `numpy` and loses columns first. The reason that `make_reduction` points to the 1st generation reducer still is that it is faster - @ericjb has been working on the performance bottlenecks in the 2nd generation reducer.

Further, we are considering to rebase this on the `skforecast` reducers, FYI @joaquinamatrodrigo, @javierescobarortiz

**Comment by fkiraly** [comment]
FYI @yarnabrina

**Comment by fkiraly** [comment]
As a temporary hack, we could consider adding a feature where - prior to passing to the `sklearn` estimator - we convert to `pd.DataFrame` with the correct columns. I am interpreting the request to add this to the v1 reducer, `_RecursiveReducer`

**Comment by fkiraly** [comment]
I am adding a "good first issue" label, with the express caution that this is layers of legacy code that may be fiddly to modify or test. As a recipe: check where code is passed to the `sklearn` `estimator`, and do an optional convert to `pd.DataFrame` before that.

The column names for `y` and `X` can be found in `self._X_metadata` and `_self._y_metadata`, although this has to be adapted so the lags etc also get appropriate column names.

Optimally, this is rendered consistent with the 2nd generation `RecursiveReductionForecaster`.

**Comment by haidmoham** [comment]
Hi! I’m interested in working on this as my first sktime contribution. I noticed the previous attempt in #8872 was recently closed. Is this issue still available, and is the direction described above still current? If so, I’d be happy to investigate the prior attempt and open a small draft PR early for feedback.

**Comment by ooobiii** [comment]
Hi! I'd like to pick this up as my first sktime contribution. I've read through the thread including the earlier attempt in #8872 and the recipe in fkiraly's comments, so my plan follows the accepted direction: an optional __init__-level flag that converts the input to a pd.DataFrame (with proper column names for the lag features) right before estimator.fit, kept consistent with the 2nd-generation reducers. 

If haidmoham hadn't picked this, could I please be assigned? Thnaks
